client/src/unifyfs.c

```diff
@@ -1288,8 +1262,11 @@ int unifyfs_fid_open(const char* path, int flags, mode_t mode, int* outfid,
         }
 
         /* initialize global size of file from global metadata */
-        unifyfs_filemeta_t* meta = unifyfs_get_meta_from_fid(fid);
-        meta->global_size = gfattr.size;
+        ret = unifyfs_fid_update_file_meta(fid, &gfattr);
+        if (ret != UNIFYFS_SUCCESS) {
+            LOGERR("failed to update file metadata from global");
+            /* this is not a fatal error, continue */
```

Comment:
Why wouldn't this be a fatal error?  We need to know if the file is laminated on open().

Reply:
Because fid will only be valid if we already have the file open locally.

Reply:
the short answer is that the return code check really isn't needed, because we already know we have a valid fid and non-null pointer, which are the only reasons the function could return an error. I actually thought about leaving out the check, but left it because it was more consistent with other usage.

Reply:
Ok, then I would just add a comment above `unifyfs_fid_update_file_meta()` saying "we assume fid and gfattr have already been verified", and remove the check and the return code.  That simplifies the function, and removes the need for callers to verify the `unifyfs_fid_update_file_meta()` return code.

Reply:
For this particular case we know the parameters are valid. There are other cases where the function may be called with `fid == -1`, so we can't just assume all uses will be good. I can remove the check for this use though.